io/src/real_sense_2_grabber.cpp

```diff
@@ -0,0 +1,219 @@
+{
+    return pcl::PointCloud<pcl::PointXYZI>::Ptr();
+}
+
+pcl::PointCloud<pcl::PointXYZRGB>::Ptr RealSense2Grabber::convertRGBDepthToPointXYZRGB(const rs2::points & points, rs2::video_frame & rgb)
+{
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZRGB>());
+
+    auto sp = points.get_profile().as<rs2::video_stream_profile>();
+    cloud->width = sp.width();
+    cloud->height = sp.height();
+    cloud->is_dense = false;
+    cloud->points.resize(points.size());
+
+    auto ptr = points.get_vertices();
+    auto uvptr = points.get_texture_coordinates();
+
+    uint8_t r, g, b;
+    for (auto& p : cloud->points)
+    {
+        p.x = ptr->x;
+        p.y = -1 * ptr->y;
+        p.z = ptr->z;
+
+        std::tie(r, g, b) = get_texcolor(rgb, uvptr->u, uvptr->v);
+
+        p.r = r;
+        p.g = g;
+        p.b = b;
+        p.a = 255;
+
+        ptr++;
+        uvptr++;
+    }
+
+    return cloud;
+}
+
+pcl::PointCloud<pcl::PointXYZRGBA>::Ptr RealSense2Grabber::convertRGBADepthToPointXYZRGBA(const rs2::points & points, rs2::video_frame & rgb)
+{
+    pcl::PointCloud<pcl::PointXYZRGBA>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZRGBA>());
+
+    auto sp = points.get_profile().as<rs2::video_stream_profile>();
+    cloud->width = sp.width();
+    cloud->height = sp.height();
+    cloud->is_dense = false;
+    cloud->points.resize(points.size());
+
+    auto ptr = points.get_vertices();
+    auto uvptr = points.get_texture_coordinates();
+
+    uint8_t r, g, b;
+    for (auto& p : cloud->points)
+    {
+        p.x = ptr->x;
+        p.y = -1 * ptr->y;
+        p.z = ptr->z;
+
+        std::tie(r, g, b) = get_texcolor(rgb, uvptr->u, uvptr->v);
+
+        p.r = r;
+        p.g = g;
+        p.b = b;
+        p.a = 255;
+
+        ptr++;
+        uvptr++;
+    }
+
+    return cloud;
+}
+
+std::tuple<uint8_t, uint8_t, uint8_t> RealSense2Grabber::get_texcolor(rs2::video_frame &texture, float u, float v)
+{
+    auto ptr = dynamic_cast<rs2::video_frame*>(&texture);
+
+    const int w = ptr->get_width(), h = ptr->get_height();
+    int x = std::min(std::max(int(u*w + .5f), 0), w - 1);
+    int y = std::min(std::max(int(v*h + .5f), 0), h - 1);
+    int idx = x * ptr->get_bits_per_pixel() / 8 + y * ptr->get_stride_in_bytes();
+    const auto texture_data = reinterpret_cast<const uint8_t*>(ptr->get_data());
+    return std::make_tuple(texture_data[idx], texture_data[idx + 1], texture_data[idx + 2]);
+}
+
+void RealSense2Grabber::threadFunction()
+{
+    while (!quit)
+    {
+        boost::unique_lock<boost::mutex> lock(mutex);
+
+        // Wait for the next set of frames from the camera
+        auto frames = pipe.wait_for_frames();
+
+        auto depth = frames.get_depth_frame();
+
+        // Generate the pointcloud and texture mappings
+        auto points = pc.calculate(depth);
+
+        // Tell pointcloud object to map to this color frame
+        auto rgb = frames.get_color_frame();
+
```

Comment:
librealsense2 provide <code>rs2::frameset::get_infrared_frame()</code> since v2.10.0.
You can easily to get infrared frame like any other frames.
Please see this [pull request](https://github.com/IntelRealSense/librealsense/pull/1122).

Reply:
I've added the IR frame and processing.  Thanks for adding the handy function 🥇

Reply:
Yes, the code was updated to use the `frames.get_infrared_frame();` convenience method

see: https://github.com/PointCloudLibrary/pcl/pull/2214/files#diff-2c7a8b41ca6e6bcff66df4b94d4d76b0R201